### PR TITLE
Implement fixed term loan adjustment

### PR DESCRIPTION
**Pull request by skalermo**

### CODE REVIEW

File: contracts/truefi2/LoanFactory2.sol [hunk 1]
```diff
@@ -53,12 +66,24 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
         lender = _lender;
         admin = msg.sender;
         liquidator = _liquidator;
+
+        adjustmentCoefficient = 25;
     }
 
     function setAdmin() external {
```
Comment (vanruch) [inline contracts/truefi2/LoanFactory2.sol:73]: Btw we can remove this function

File: contracts/truefi2/LoanFactory2.sol [hunk 2]
```diff
@@ -53,12 +66,24 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
...
     }
 
+    function setAdjustmentCoefficient(uint256 newCoefficient) external {
+        require(msg.sender == admin, "LoanFactory: Only admin can set adjustment coefficient");
+        adjustmentCoefficient = newCoefficient;
+        emit AdjustmentCoefficientChanged(newCoefficient);
+    }
+
+    function fixTermLoanAdjustment(uint256 term) public view returns (uint256) {
+        return term.div(30 days).mul(adjustmentCoefficient);
```
Comment (vanruch) [inline contracts/truefi2/LoanFactory2.sol:84]:
```suggestion
        return term.mul(adjustmentCoefficient).div(30 days);
```

File: contracts/truefi2/LoanFactory2.sol [hunk 3]
```diff
@@ -53,12 +66,24 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
...
         admin = 0x16cEa306506c387713C70b9C1205fd5aC997E78E;
     }
 
+    function setAdjustmentCoefficient(uint256 newCoefficient) external {
+        require(msg.sender == admin, "LoanFactory: Only admin can set adjustment coefficient");
+        adjustmentCoefficient = newCoefficient;
+        emit AdjustmentCoefficientChanged(newCoefficient);
+    }
+
+    function fixTermLoanAdjustment(uint256 term) public view returns (uint256) {
```
Comment (vanruch) [inline contracts/truefi2/LoanFactory2.sol:83]: `fixedTermLoanAdjustment`

File: contracts/truefi2/LoanFactory2.sol (hunk 2 above)
Comment (skalermo) [reply contracts/truefi2/LoanFactory2.sol:84]: This is intentional, as it allows to change adjustment non-continuously

File: test/truefi2/LoanFactory2.test.ts [hunk 4]
```diff
@@ -47,38 +46,80 @@ describe('LoanFactory2', () => {
...
+      await factory.setAdjustmentCoefficient(25)
+    })
+
+    ;[
+      [0, 0],
+      [30 * DAY - 1, 0],
+      [30 * DAY, 25],
+      [60 * DAY - 1, 25],
+      [60 * DAY, 50],
+    ].map(([term, adjustment]) =>
```
Comment (barrutko) [inline test/truefi2/LoanFactory2.test.ts:119]: Add one or two longer cases. For example 3,5 months and 6 months.

File: contracts/truefi2/TrueRateAdjuster.sol [hunk 5]
```diff
@@ -4,15 +4,22 @@ pragma solidity 0.6.10;
...
 
     uint256 constant MAX_RATE_CAP = 50000;
     uint8 constant MAX_CREDIT_SCORE = 255;
 
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
```
Comment (vanruch) [inline contracts/truefi2/TrueRateAdjuster.sol:22]: This contract is not initializable so this comment is lying :)

File: contracts/truefi2/LoanFactory2.sol [hunk 6]
```diff
@@ -47,18 +54,24 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
...
         liquidator = _liquidator;
+        rateAdjuster = _rateAdjuster;
     }
 
     function setAdmin() external {
         admin = 0x16cEa306506c387713C70b9C1205fd5aC997E78E;
     }
 
+    function fixedTermLoanAdjustment(uint256 term) public view returns (uint256) {
+        return term.div(30 days).mul(rateAdjuster.fixedTermLoanAdjustmentCoefficient());
```
[the remark this answers is not in the document]
Comment (yuchenlintt) [reply contracts/truefi2/LoanFactory2.sol:72]: Once that's done then this PR LGTM. I'll approve now so you won't need another back-and-forth review cycle.

**Review by yuchenlintt — APPROVED** [review]

**Comment by skalermo** [comment]
Closing in favor of #836.

**Pull request closed without merging**